server/vulnerabilities/msrc/parsed/product.go

```diff
@@ -12,12 +15,53 @@ import (
 // (if any) and its version (if any).
 type Product string
 
+type Products map[string]Product
+
+var ErrNoMatch = errors.New("no product matches")
+
+func (p Products) GetMatchForOS(ctx context.Context, os fleet.OperatingSystem) (string, error) {
+	var dvMatch, noDvMatch string
+
+	for pID, p := range p {
```

Comment:
Nit: TIL you can redefine `p` like this.

Reply:
oy good catch, that's confusing